tests/scripts/all.sh

```diff
@@ -2710,20 +2711,37 @@ component_test_variable_ssl_in_out_buffer_len () {
 }
 
 component_test_variable_ssl_in_out_buffer_len_CID () {
-    msg "build: MBEDTLS_SSL_VARIABLE_BUFFER_LENGTH and MBEDTLS_SSL_DTLS_CONNECTION_ID enabled (ASan build)"
-    scripts/config.py set MBEDTLS_SSL_VARIABLE_BUFFER_LENGTH
+    msg "build: MBEDTLS_SSL_DTLS_CONNECTION_ID (standard) enabled (ASan build)"
+    scripts/config.py set MBEDTLS_SSL_DTLS_CONNECTION_ID
+
+    CC=gcc cmake -D CMAKE_BUILD_TYPE:String=Asan .
+    make
+
+    msg "test: MBEDTLS_SSL_DTLS_CONNECTION_ID (standard)"
+    make test
+
+    msg "test: ssl-opt.sh, MBEDTLS_SSL_DTLS_CONNECTION_ID (standard) enabled"
+    tests/ssl-opt.sh
+
+    msg "test: compat.sh, MBEDTLS_SSL_DTLS_CONNECTION_ID (standard) enabled"
+    tests/compat.sh
+}
+
+component_test_variable_ssl_in_out_buffer_len_CID_legacy () {
```

Comment:
The component name does not reflect its content, we should remove the `in_out_buffer_len` part.

Also, the component is failing on the CI (both internal and open, so it's unlikely to be just bad luck).